homeassistant/components/roborock/config_flow.py

```diff
@@ -137,6 +139,25 @@ async def async_step_code(
             errors=errors,
         )
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle a flow started by a dhcp discovery."""
+        roborock_entries = self.hass.config_entries.async_entries(DOMAIN)
+        device_registry = dr.async_get(self.hass)
+        for entry in roborock_entries:
+            device_entries = dr.async_entries_for_config_entry(
+                device_registry, config_entry_id=entry.entry_id
+            )
+            for device in device_entries:
+                for connection in device.connections:
+                    if (
+                        connection[0] == "mac"
+                        and connection[1].replace(":", "") == discovery_info.macaddress
```

Comment:
could also invert this and use the existing `format_mac` here and when creating the device in the first place. That is, my impression is the device registry has a specific format so it would be good to be defensive in case the roborock format is not that format. (e.g. case is wrong)